### PR TITLE
Add `atmos aws` commands. Add `atmos aws eks update-kubeconfig` command

### PR DESCRIPTION
**Pull request by aknysh**
## what
* Add `atmos aws` commands
* Add `atmos aws eks update-kubeconfig` command

## why
* Execute `aws` CLI commands using `atmos` context (`atmos.yaml` CLI config and component/stack configurations)
* Downloading `kubeconfig` from an EKS cluster using `atmos
* Allow using this functionality in Terraform by implementing a new data source in the `terraform-provider-utils` provider (this will be added in a PR in https://github.com/cloudposse/terraform-provider-utils repo)

```
`atmos aws eks update-kubeconfig`  command can be executed in three different ways:

1. If all the required parameters (cluster name and AWS profile/role) are provided on the command-line, 
then 'atmos' executes the command without requiring atmos CLI config and context.
For example: atmos aws eks update-kubeconfig --profile=<profile> --name=<cluster_name>

2. If 'component' and 'stack' are provided on the command-line, 
   then 'atmos' executes the command using atmos CLI config and stack's context by searching for the following settings:
  - 'components.helmfile.cluster_name_pattern' in 'atmos.yaml' CLI config (and calculates the '--name' parameter using the pattern)
  - 'components.helmfile.helm_aws_profile_pattern' in 'atmos.yaml' CLI config (and calculates the '--profile' parameter using the pattern)
  - 'components.helmfile.kubeconfig_path' in 'atmos.yaml' CLI config
  - the variables for the component in the provided stack
  - 'region'' from the variables for the component in the stack
For example: atmos aws eks update-kubeconfig <component> -s <stack>

3. Combination of the above. Provide a component and a stack, and override other parameters on the command line:
For example: atmos aws eks update-kubeconfig <component> -s <stack> --kubeconfig=<path_to_kubeconfig> --region=<region>
```

## references
* https://docs.aws.amazon.com/cli/latest/reference/eks/update-kubeconfig.html

### CODE REVIEW

File: cmd/aws.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+	"github.com/spf13/cobra"
+)
+
+// awsCmd executes 'aws' CLI commands
+var awsCmd = &cobra.Command{
+	Use:                "aws",
+	Short:              "Execute 'aws' commands",
+	Long:               `This command executes 'aws'' CLI commands`,
```
Comment (nitrocode) [inline cmd/aws.go:11]:
```suggestion
	Long:               `This command executes 'aws' CLI commands`,
```

File: cmd/aws_eks.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+	"github.com/spf13/cobra"
+)
+
+// awsCmd executes 'aws eks' CLI commands
+var awsEksCmd = &cobra.Command{
+	Use:                "eks",
+	Short:              "Execute 'aws eks' commands",
+	Long:               `This command executes 'aws eks'' CLI commands`,
```
Comment (nitrocode) [inline cmd/aws_eks.go:11]:
```suggestion
	Long:               `This command executes 'aws eks' CLI commands`,
```

File: cmd/aws_eks_update_kubeconfig.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+then 'atmos' executes the command without requiring atmos CLI config and context.
+For example: atmos aws eks update-kubeconfig --profile=<profile> --name=<cluster_name>
+
+2. If 'component' and 'stack' are provided on the command-line, 
+   then 'atmos' executes the command using atmos CLI config and stack's context by searching for the following settings:
+  - 'components.helmfile.cluster_name_pattern' in 'atmos.yaml' CLI config (and calculates the '--name' parameter using the pattern)
+  - 'components.helmfile.helm_aws_profile_pattern' in 'atmos.yaml' CLI config (and calculates the '--profile' parameter using the pattern)
+  - 'components.helmfile.kubeconfig_path' in 'atmos.yaml' CLI config
+  - the variables for the component in the provided stack
+  - 'region'' from the variables for the component in the stack
```
Comment (nitrocode) [inline cmd/aws_eks_update_kubeconfig.go:26]:
```suggestion
  - 'region' from the variables for the component in the stack
```

**Review by nitrocode — DISMISSED** [review]
how come there aren't any tests?

**Review by Nuru — CHANGES_REQUESTED** [review]
It should be an error to specify both `--profile` and `--role-arn` . It also looks to me like if you specify `--role-arn` but not `--cluster` then the resulting `aws` command will include both `--profile` and `--role-arn`, which is wrong. `--role-arn` on the command line should suppress `--profile` being automatically set.

**Comment by Nuru** [comment]
Feature request: One of the things I have been wanting (which may or may not be appropriate to add to this PR) is the ability to have the stacks configure a Role ARN instead of a profile for Kubeconfig. For companies using only SAML access, if Kubeconfig used Role ARNs the way Terraform uses Role ARNs then users would not need AWS config files at all. They would just log in with Leapp and everything else would happen automatically. I see you added a `--role-arn` flag to this command, so maybe you can add it to the global configuration?

**Comment by aknysh** [comment]
> 

not in this PR. Devs are waiting on this in the `utils` provider to finish other tasks

**Comment by aknysh** [comment]
> 

no tests, because to test it, we need an EKS cluster

File: internal/exec/aws_eks_update_kubeconfig.go [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+func ExecuteAwsEksUpdateKubeconfig(kubeconfigContext c.AwsEksUpdateKubeconfigContext) error {
+	// AWS profile to authenticate to the cluster
+	profile := kubeconfigContext.Profile
+
+	// To assume a role for cluster authentication, specify an IAM role ARN with this option. For example, if you created a cluster while
+	// assuming an IAM role, then you must also assume that role to connect to the cluster the first time
+	roleArn := kubeconfigContext.RoleArn
+
+	if profile != "" && roleArn != "" {
+		return errors.New(fmt.Sprintf("Eigher 'profile' or 'role-arn' can be specified, but not both. Profile: '%s'. Role ARN: '%s'", profile, roleArn))
```
Comment (Nuru) [inline internal/exec/aws_eks_update_kubeconfig.go:91]:
```suggestion
		return errors.New(fmt.Sprintf("Either 'profile' or 'role-arn' can be specified, but not both. Profile: '%s'. Role ARN: '%s'", profile, roleArn))
```

File: internal/exec/aws_eks_update_kubeconfig.go [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+			kubeconfigPath = fmt.Sprintf("%s/%s-kubecfg", c.Config.Components.Helmfile.KubeconfigPath, kubeconfigContext.Stack)
+		}
+		// `clusterName` can be overridden on the command line
+		if clusterName == "" {
+			clusterName = c.ReplaceContextTokens(context, c.Config.Components.Helmfile.ClusterNamePattern)
+		}
+		// `profile` can be overridden on the command line
+		if profile == "" {
+			profile = c.ReplaceContextTokens(context, c.Config.Components.Helmfile.HelmAwsProfilePattern)
+		}
```
Comment (Nuru) [inline internal/exec/aws_eks_update_kubeconfig.go:170]: Avoid any possible errors trying to derive a profile name if the profile is not going to be used anyway. 

```suggestion
		if profile == "" && roleArn == "" {
			profile = c.ReplaceContextTokens(context, c.Config.Components.Helmfile.HelmAwsProfilePattern)
		}
```

File: internal/exec/helmfile.go [hunk 6]
```diff
@@ -86,7 +86,7 @@ func ExecuteHelmfile(cmd *cobra.Command, args []string) error {
 	clusterName := c.ReplaceContextTokens(context, c.Config.Components.Helmfile.ClusterNamePattern)
 	color.Cyan(fmt.Sprintf("Downloading kubeconfig from the cluster '%s' and saving it to %s\n\n", clusterName, kubeconfigPath))
 
-	err = execCommand("aws",
+	err = ExecuteShellCommand("aws",
```
Comment (Nuru) [inline internal/exec/helmfile.go:89]: Can we easily allow `--role-arn` for the `helmfile` command?

**Review by Nuru — CHANGES_REQUESTED** [review]

File: internal/exec/helmfile.go (hunk 6 above)
Comment (aknysh) [reply internal/exec/helmfile.go:89]: let's not do it in this PR

**Review by Nuru — APPROVED** [review]

**Pull request merged**